LoopKit/Service/Remote/Actions/RemoteBolusAction.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  RemoteBolusAction.swift
+//  LoopKit
+//
+//  Created by Bill Gestrich on 12/25/22.
+//  Copyright © 2022 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+public struct RemoteBolusAction: Codable {
+    
+    public let amountInUnits: Double
+    
+    public init(amountInUnits: Double) {
+        self.amountInUnits = amountInUnits
+    }
+}
+
+extension RemoteBolusAction {
```

Comment:
Putting the validation logic needed to enact an action, in the action itself. I wrestled with keeping this logic in Loop instead, but I kind of like keeping the validation logic with the model.

Reply:
Why are these in LoopKit at all? If it's for sharing types with plugins, maybe that would be necessary. If it's for sharing types between Loop and LCG, maybe pause on that for now, and keep them separate.  Human input validation (LCG-only concern), network decoding and validation, and Loop's other validation concerns (like max bolus, max temp), are all separate things. Guessing there might be other differences in concerns between them that might make the types different too.  Maybe Loop wants to have a received time?

Reply:
If these types are *strictly* for representing a Swift version of the network encoding of these actions, then I guess that could be a shared concern.

Reply:
I think I'm following the suggested separation here. These models are to include all the information needed to enact an action in Loop. For remote 2.0, they will be used for the network encoding/decoding both from the plugins and Caregiver. 

I can move the validation pieces out of here and put into the Loop target. That does leave the question of where to put these things to keep them unit testable as they are. Making an extension, like here, keeps it testable without involving the DeviceDataManager test setup or dedicated validation structures for each action (i.e. RemoteBolusValidator..). My inclination is to move the extensions to the actions to the Loop target to provide some separation. But I can consider a validator or wrapping these actions into Loop domain actions if something more dedicated is desired. Note that in remote 2.0, the plan is for a `RemoteCommand` to hold onto a `RemoteAction` and other things like the `RemoteCommandState`.

Reply:
For context, here is where I was planning to take things in remote 2.0 wrt actions. The RemoteCommand holds onto the action. Everything else in the RemoteCommand is specific to command management and lifecycle. 

Rather than juggling a separate hook to the service that generated the command, the RemoteCommand itself has methods to interact with its service internally.. ex: `RemoteCommand.markSuccess()` rather than `service.markSuccess(command)`

```

public protocol RemoteCommand {
    
    var id: String {get}
    var action: RemoteAction {get}
    var status: RemoteCommandStatus {get}
    //TODO: Add creation date
    
    /*
     TODO: Should we support "future" commands?
     TODO: Should the expiration date be in here?
     TODO: Should creationDate be in here?
     */
    
    func checkValidity() throws
    
    func markInProgress() async throws
    func markError(_ error: Error) async throws
    func markSuccess() async throws
}
```

Reply:
I think moving them to Loop sounds good. Yes, you will have a similar type in LCG, with similar fields, but I think this is one case where it's ok to avoid DRY. The responsibilities and things you will do with those types are going to be different.

Thanks for the context on eventual plans of wrapping RemoteAction with RemoteCommand. What do you think about RemoteAction -> Action?  If a RemoteCommand has an action, the "remote" part can be implicit (if "actions" are things that could theoretically be triggered locally, by, say, a siri command), or could be namespaced by RemoteCommand (if they are not things that could be reused locally).

Reply:
I think making the Actions decoupled from the remote aspect makes sense. I've dropped the Remote verbiage from actions for now.